test/plugin/base_test.rb

```diff
@@ -338,6 +338,81 @@ def test_structured_payload_log
     end
   end
 
+  def test_stackdriver_trace_format_compliant_trace
```

Comment:
The logic in these three tests are almost identical. They could be combined and use something like:

```
      {
        config: APPLICATION_DEFAULT_CONFIG,
        input_trace: XXX,
        expected_trace: XXX
      },
```

This way, it's easier to maintain the shared logic and also execute all tests related to `trace` with a single command. It also makes it easier to add more test cases.

BTW, we should also test cases like below in combination of different config values:
- There's no `logging.googleapis.com/trace` field in payload.
- `logging.googleapis.com/trace` field in payload is an empty string.
- A custom `trace_key` is used instead of the default trace key. (Reference: `trace_key` is configurable: https://github.com/GoogleCloudPlatform/fluent-plugin-google-cloud/blob/master/lib/fluent/plugin/out_google_cloud.rb#L284)